common/autoware_ad_api_msgs/routing/srv/SetRoute.srv

```diff
@@ -0,0 +1,7 @@
+std_msgs/Header header
+geometry_msgs/Pose[<=1] start
+geometry_msgs/Pose goal
```

Comment:
(Just a comment.)
I guess we might need to add a type (such as approximate or fixed) or an acceptable error for the `goal`.
It's for adjusting the goal by Autoware when the original goal isn't available.

Reply:
@mitsudome-r @yukkysaito @TakaHoribe What do you think about this?

Reply:
Good point, but I think we should first investigate how the given error can be achieved in the planning stack. If we can set an acceptable error for the goal in the API, but if there is no module in Autoware that ensures the error, it would be confusing for the user.

Reply:
Yes, exactly. I meant we have to improve this API in the future.
So it's okay to use this message for now.

Reply:
I think it is good to introduce something like goal type in this PR.
sample of goal type:
- hard_goal_constraints(don't allow refine goal)
  - distance_range
  - yaw_range
- soft_goal_constraints(allow refine goal)
  - foo

Reply:
Discussion result with @isamu-takagi @mitsudome-r @yukkysaito

- We need some types or error range in the future.
- But since the requirements aren't clear enough now, we can start with the current simple message definition.
- We can switch the behavior of whether to allow modifying the goal or not, by changing the parameters of the Planning components.